core/gdxsv/gdxsv_network.cpp

```diff
@@ -26,12 +26,68 @@ bool TcpClient::Connect(const char *host, int port) {
     memcpy(&addr.sin_addr, host_entry->h_addr_list[0], host_entry->h_length);
 #endif
     addr.sin_port = htons(port);
-    set_recv_timeout(new_sock, 5000);
+    
+    fd_set setW, setE;
+    struct timeval timeout = {5,0};
+    int res;
+    
+    auto set_blocking_mode = [](const int &socket, bool is_blocking) {
+#ifdef _WIN32
+        u_long flags = is_blocking ? 0 : 1;
+        ioctlsocket(socket, FIONBIO, &flags);
+#else
+        const int flags = fcntl(socket, F_GETFL, 0);
+        fcntl(socket, F_SETFL, is_blocking ? flags ^ O_NONBLOCK : flags | O_NONBLOCK);
+#endif
+    };
+    
+    set_blocking_mode(new_sock, false);
+    
+    ::connect(new_sock, (const sockaddr *) &addr, sizeof(addr));
```

Comment:
don't ignore return value of `connect()`

Reply:
done, always check the return for `connect` now, will fail back to blocking mode is there is no error